t/scripts/submit_as.py

```diff
@@ -34,6 +34,8 @@ def main():
     userid = int(sys.argv[1])
     os.environ["FLUX_HANDLE_USERID"] = str(userid)
 
+    urgency = sys.argv[2] if "urgency" in sys.argv[2] else None
```

Comment:
Since `submit-as.py` takes any `flux-mini run` option, it won't be guaranteed that `--urgency` will be the second option. Therefore, I think it might be better (i.e. less surprising) if `sys.argv[2:]` was iterated and any option matching `--urgency` was passed down to `flux job submit`.

Also, the code could be simplified if you just append to the `submitcmd` list if needed.

e.g. something like (untested):


```diff
diff --git a/t/scripts/submit_as.py b/t/scripts/submit_as.py
index 4b22c92..6bd45b9 100644
--- a/t/scripts/submit_as.py
+++ b/t/scripts/submit_as.py
@@ -34,7 +34,10 @@ def main():
     userid = int(sys.argv[1])
     os.environ["FLUX_HANDLE_USERID"] = str(userid)
 
-    urgency = sys.argv[2] if "urgency" in sys.argv[2] else None
+    submitcmd = [ "flux", "job", "submit", "--flags=signed" ]
+    for arg in sys.argv[2:]:
+        if "urgency" in arg:
+            submitcmd.append(arg)
 
     minicmd = [
         "flux",
@@ -48,10 +51,6 @@ def main():
 
     signedJ = SecurityContext().sign_wrap_as(userid, jobspec, mech_type="none")
 
-    if urgency is not None:
-        submitcmd = ["flux", "job", "submit", urgency, "--flags=signed"]
-    else:
-        submitcmd = ["flux", "job", "submit", "--flags=signed"]
     jobid = run_process(submitcmd, input=signedJ)
     print(jobid)
```
Good catch by the way!

Reply:
Thanks @grondo for reviewing and for the suggested improvement! I didn't know you could just append the `urgency` argument to the end of `submitcmd`, so thank you for pointing that out. This solution is much cleaner and did result in passing tests, so I just force pushed a change that implemented this suggestion and rebased to catch up after some of the PR's that have landed before this one. I've also taken it out of [WIP].